### DISCUSSION TITLE
Caught signal

### DISCUSSION THREAD

**Discussion by zakorra** [nxrighthere/UnrealCLR#128] [Help] [answered]
Hi,

first of all a big thank you for this great plugin.
I was looking forward to do unreal with C# and since now I didn't start
since unreal development was not possible in C#. 
---

I followed the documentation and has setup everything as described.
(Hint: Unreal Editor is working perfect on my mac and I can compile and debug c++ projects).

When I try to start the test scene I get the error below.

Somebody have some advice what I am doing wrong?
Seems that some libs cannot be found...
Many thanks in advance for helping

![Bildschirmfoto 2021-05-03 um 10 53 29](https://user-images.githubusercontent.com/17723680/116857876-cd0a1e00-abfd-11eb-8141-975fc8d4d55c.png)

**Comment by nxrighthere** [comment, marked as answer]
Exceptions on Linux and macOS terminates the editor, please, attach a debugger to it and check stack trace.

**Comment by zakorra** [comment]
thx for the hint. I started the Unreal Editor from Rider and received the error that the Material cannot be loaded:
_Failed to load '/Game/Tests/BasicMaterial': Can't find file._

I can fixed this temporary  by:
- opening / saving '/Game/Tests/BasicMaterial'
- saving the Test-Map


nevertheless, the test content look weird after playing:
Before Play:
<img width="1347" alt="Bildschirmfoto 2021-05-04 um 20 08 09" src="https://user-images.githubusercontent.com/17723680/117049594-9b37ab00-ad14-11eb-98ab-1b9eab987ad8.png">

After Play:
<img width="1358" alt="Bildschirmfoto 2021-05-04 um 20 06 01" src="https://user-images.githubusercontent.com/17723680/117049282-3b410480-ad14-11eb-8047-33eb383c1ad0.png">


On next start of the Unreal-Editor the Test-Map crash again.

----------
Output

Log          LogTemp              Repeating last play command: Selected Viewport
Log          LogPlayLevel         PlayLevel: No blueprints needed recompiling
Log          PIE                  New page: PIE session: Tests (4 May 2021, 22:10:59)
Log          LogOnline            OSS: Creating online subsystem instance for: NULL
Log          LogOnline            OSS: TryLoadSubsystemAndSetDefault: Loaded subsystem for module [NULL]
Log          LogPlayLevel         Creating play world package: /Game/Tests/UEDPIE_0_Tests
Display      LogMetal             Got a bulk data texture, with 1 mips
Display      LogMetal             Got a bulk data texture, with 1 mips
Display      LogMetal             Got a bulk data texture, with 1 mips
Log          LogPlayLevel         PIE: StaticDuplicateObject took: (0.001817s)
Log          LogAIModule          Creating AISystem for world Tests
Display      LogUnrealCLR         Log: Framework loaded succesfuly for /Users/Shared/Unreal/CetaUnreal/Managed/Tests/UnrealEngine.Tests.dll
Log          LogPlayLevel         PIE: World Init took: (0.052424s)
Log          LogPlayLevel         PIE: Created PIE world by copying editor world from /Game/Tests/Tests.Tests to /Game/Tests/UEDPIE_0_Tests.Tests (0.054602s)
Log          LogUObjectHash       Compacting FUObjectHashTables data took   0.37ms
Display      LogAudio             Creating Audio Device:                 Id: 2, Scope: Unique, Realtime: True
Display      LogAudioMixer        Audio Mixer Platform Settings:
Display      LogAudioMixer        	Sample Rate:						  48000
Display      LogAudioMixer        	Callback Buffer Frame Size Requested: 1024
Display      LogAudioMixer        	Callback Buffer Frame Size To Use:	  1024
Display      LogAudioMixer        	Number of buffers to queue:			  2
Display      LogAudioMixer        	Max Channels (voices):				  32
Display      LogAudioMixer        	Number of Async Source Workers:		  0
Display      LogAudio             AudioDevice MaxSources: 32
Display      LogAudio             Using built-in audio occlusion.
Display      LogAudioMixer        Initializing audio mixer.
Display      LogAudioMixer        Using Audio Device Unknown
Display      LogAudioMixer        Initializing Sound Submixes...
Display      LogAudioMixer        Creating Master Submix 'MasterSubmixDefault'
Display      LogAudioMixer        Creating Master Submix 'MasterReverbSubmixDefault'
Display      LogAudioMixer        Creating Master Submix 'MasterEQSubmixDefault'
Display      LogAudioMixer        Output buffers initialized: Frames=1024, Channels=2, Samples=2048
Display      LogAudioMixer        Starting AudioMixerPlatformInterface::RunInternal()
Log          LogInit              FAudioDevice initialized.
Display      LogAudio             Audio Device (ID: 2) registered with world 'Tests'.
Log          LogLoad              Game class is 'TestsGameMode_C'
Log          LogWorld             Bringing World /Game/Tests/UEDPIE_0_Tests.Tests up for play (max tick rate 0) at 2021.05.04-20.11.00
Log          LogWorld             Bringing up level for play took: 0.000454
Log          LogOnline            OSS: Creating online subsystem instance for: :Context_2
Log          PIE                  Server logged in
Log          PIE                  Play in editor total start time 0,082 seconds.
Display      LogUnrealManaged     Log: Hello, Unreal Engine!
Warning      LogLinker            Failed to load '/Game/Tests/BasicMaterial': Can't find file.
Warning      LogLinker            Failed to load '/Game/Tests/BasicMaterial': Can't find file.
Warning      LogUObjectGlobals    Failed to find object 'Material None./Game/Tests/BasicMaterial'
Log          LogMac               === Critical error: ===
Log          LogMac               SIGABRT: abort() called
Log          LogMac               
Log          LogMac               0x20312946 libsystem_kernel.dylib!__pthread_kill()   [UnknownFile])
Log          LogMac               0x20296411 libsystem_c.dylib!abort()   [UnknownFile])
Log          LogMac               0xee89e91e libcoreclr.dylib!PROCAbort()   [UnknownFile])
Log          LogMac               0xee471a87 libcoreclr.dylib!PROCEndProcess(void*, unsigned int, int)   [UnknownFile])
Log          LogMac               0xee8a148c libcoreclr.dylib!UnwindManagedExceptionPass1(PAL_SEHException&, _CONTEXT*) (.cold.4)   [UnknownFile])
Log          LogMac               0xee6c093e libcoreclr.dylib!UnwindManagedExceptionPass1(PAL_SEHException&, _CONTEXT*)   [UnknownFile])
Log          LogMac               0xee6c09c6 libcoreclr.dylib!DispatchManagedException(PAL_SEHException&, bool)   [UnknownFile])
Log          LogMac               0xee624134 libcoreclr.dylib!IL_Throw(Object*)   [UnknownFile])
Log          LogMac               
Log          LogMac               
Log          LogMac               
Log          LogMac               
Log          LogMac               
Log          LogMac               
Log          LogMac               0xee298ffd UE4Editor-UnrealCLR.dylib!UnrealCLR::PrePhysicsTickFunction::ExecuteTick(float, ELevelTick, ENamedThreads::Type, TRefCountPtr<FGraphEvent> const&)   [UnknownFile])
Log          LogMac               0x0a77fd8b UE4Editor-Engine.dylib!FTickFunctionTask::DoTask(ENamedThreads::Type, TRefCountPtr<FGraphEvent> const&)   [UnknownFile])
Log          LogMac               0x0a77f5b4 UE4Editor-Engine.dylib!TGraphTask<FTickFunctionTask>::ExecuteTask(TArray<FBaseGraphTask*, TSizedDefaultAllocator<32> >&, ENamedThreads::Type)   [UnknownFile])
Log          LogMac               0x07121466 UE4Editor-Core.dylib!FNamedTaskThread::ProcessTasksNamedThread(int, bool)   [UnknownFile])
Log          LogMac               0x0711fa80 UE4Editor-Core.dylib!FNamedTaskThread::ProcessTasksUntilQuit(int)   [UnknownFile])
Log          LogMac               0x0711cd6d UE4Editor-Core.dylib!FTaskGraphImplementation::WaitUntilTasksComplete(TArray<TRefCountPtr<FGraphEvent>, TInlineAllocator<4u, TSizedDefaultAllocator<32> > > const&, ENamedThreads::Type)   [UnknownFile])
Log          LogMac               0x0a777e9e UE4Editor-Engine.dylib!FTickTaskSequencer::ReleaseTickGroup(ETickingGroup, bool)   [UnknownFile])
Log          LogMac               0x0a77199a UE4Editor-Engine.dylib!FTickTaskManager::RunTickGroup(ETickingGroup, bool)   [UnknownFile])
Log          LogMac               0x09bd997a UE4Editor-Engine.dylib!UWorld::Tick(ELevelTick, float)   [UnknownFile])
Log          LogMac               0x127649a0 UE4Editor-UnrealEd.dylib!UEditorEngine::Tick(float, bool)   [UnknownFile])
Log          LogMac               0x1333c6d8 UE4Editor-UnrealEd.dylib!UUnrealEdEngine::Tick(float, bool)   [UnknownFile])
Log          LogMac               0x06d90e1e UE4Editor!FEngineLoop::Tick()   [UnknownFile])
Log          LogMac               0x06d9f64f UE4Editor!GuardedMain(char16_t const*)   [UnknownFile])
Log          LogMac               0x06dbd52f UE4Editor!-[UE4AppDelegate runGameThread:]   [UnknownFile])
Log          LogMac               0x073397e1 UE4Editor-Core.dylib!-[FCocoaGameThread main]   [UnknownFile])
Log          LogMac               0x211bb977 Foundation!__NSThread__start__()   [UnknownFile])
Log          LogMac               0x20341954 libsystem_pthread.dylib!_pthread_start()   [UnknownFile])
Log          LogMac               0x2033d4a7 libsystem_pthread.dylib!thread_start()   [UnknownFile])
Log          LogMac               
Log          LogExit              Executing StaticShutdownAfterError
}}

**Reply by nxrighthere** [reply]
It seems that installation was done improperly, try to re-run the [installation tool](https://github.com/nxrighthere/UnrealCLR#auto). If it doesn't help, please, check your drive for errors and bad sectors.
